### ISSUE TITLE
Provide data quality dumps for a custom polygon

### ISSUE THREAD

**Issue by ramyaragupathy** [hotosm/galaxy-api#75]
An API endpoint that filters out data quality issues for user specified boundaries

**Data Source:** Underpass

**Sample request body:**

```
{
“issue_type”: [bad_geometry/bad_value/incomplete_tags/all]
 "geometry": {
        "type": "Polygon",
        "coordinates": [
          [
            [
              71.08154296875,
              34.125447565116126
            ],
            [
              71.83959960937499,
              34.125447565116126
            ],
            [
              71.83959960937499,
              34.615126683462194
            ],
            [
              71.08154296875,
              34.615126683462194
            ],
            [
              71.08154296875,
              34.125447565116126
            ]
          ]
        ]
      }
}
```

**Acceptance Criteria:**
* User feeds in a custom polygon
    - Accepted GeoJSoN should be a single polygon - only one feature within the GeoJSoN FeatureCollection. On validation, throws 400 Bad  Request with a relevant error message - `Only one polygon can be provided as input`
   - To start with we are not supporting Multipolygons. On validation, throws 400 Bad Request with a relevant error message - `Multipolygons/other geometry types not supported`
    - Geometry should belong to SRID = WGS 4326, Projection WGS 1984. On validation, throws 400 Bad Request with a relevant error message - `Only WGS1984 Projection Type supported`
    - No projection conversion to be handled within API at this stage
    - Limiting the area to 5000 sq.km - similar to Tasking Manager, subject to change depending on query performance. When the geometry exceeds the agreed area limit API sends a relevant message to the frontend
    - Limiting the size of the geojson file to 2MB. Throws 400 Bad Request with a relevant error message - ``
    - API should accept only feature geometry - Throws 400 Bad Request with a relevant error message - `Only feature geometries accepted as input`
    - For polygons outside countries that are loaded within the underpass data source, the query gives an empty result.
    - For polygons that partially fit a loaded country, we return results using data from the area matching with the loaded country.

* User can select the data quality issue from a list of options
    - Required input value
    - String array 
         - bad_geometry 
         - bad_value 
         - incomplete_tags
         - All
   - Any field other than the options provided should throw an error -  400 Bad Request
  - Non-empty array should throw an error -  400 Bad Request
**Output file format:** 
**GeoJSON** 
```
{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "properties": {
“Osm_id”: int,
“Changeset_id”: int,
“Changeset_timestamp”: timestamp,
“Issue_type”:string,
“User_name”: string,
“hashtags”: []
},
      "geometry": {
        "type": "Point",
        "coordinates": [
          97.734375,
          58.26328705248601
        ]
      }
    }
  ]
}
```
**CSV**
OSM ID|Changeset ID|latitude|longitude|timestamp|Issue type|user name|hashtags

**Comment by itskshitiz321** [comment]
Related to #24

**Comment by itskshitiz321** [comment]
fixed by PR #106
